code/modules/atmospherics/environmental/LINDA_turf_tile.dm

```diff
@@ -175,64 +174,48 @@
 	for(var/t in adjacent_turfs)
 		var/turf/open/enemy_tile = t
 
-		if(fire_count > enemy_tile.current_cycle)
-			enemy_tile.archive()
-
-		/******************* GROUP HANDLING START *****************************************************************/
-
-			var/should_share_air = FALSE
-			var/datum/gas_mixture/enemy_air = enemy_tile.air
-			if(enemy_tile.excited)
-				//cache for sanic speed
-				var/datum/excited_group/enemy_excited_group = enemy_tile.excited_group
-				if(our_excited_group)
-					if(enemy_excited_group)
-						if(our_excited_group != enemy_excited_group)
-							//combine groups (this also handles updating the excited_group var of all involved turfs)
-							our_excited_group.merge_groups(enemy_excited_group)
-							our_excited_group = excited_group //update our cache
-						should_share_air = TRUE
-					else
-						if((recently_active == 1 && enemy_tile.recently_active == 1) || our_air.compare(enemy_air))
-							our_excited_group.add_turf(enemy_tile) //add enemy to our group
-							should_share_air = TRUE
+		if(fire_count <= enemy_tile.current_cycle)
+			continue
+		enemy_tile.archive()
+
+	/******************* GROUP HANDLING START *****************************************************************/
+
+		var/should_share_air = FALSE
+		var/datum/gas_mixture/enemy_air = enemy_tile.air
+
+		//cache for sanic speed
+		var/datum/excited_group/enemy_excited_group = enemy_tile.excited_group
+
+		if(our_excited_group && enemy_excited_group)
+			if(our_excited_group != enemy_excited_group)
+				//combine groups (this also handles updating the excited_group var of all involved turfs)
+				our_excited_group.merge_groups(enemy_excited_group)
+				our_excited_group = excited_group //update our cache
+			should_share_air = TRUE
+
+		else if(our_air.compare(enemy_air))
+			if(!enemy_tile.excited)
+				SSair.add_to_active(enemy_tile)
+			var/datum/excited_group/EG = our_excited_group || enemy_excited_group || new
+			if(!our_excited_group)
+				EG.add_turf(src)
+			if(!enemy_excited_group)
+				EG.add_turf(enemy_tile)
```

Comment:
these 5 lines are the most likely to harm performance a tad - they're the big reason this needs profiling done